subconscious/model.py

```diff
@@ -314,17 +287,51 @@ def get_index_key(cls, column_name):
                 first_iteration = False
             else:
                 result_set = result_set.intersection(temp_set)
+        if not kwargs:
+            for index_entry in await db.zrange(cls.get_index_key(cls._identifier_column_names[0]), 0, -1):
+                result_set.add(index_entry.split(VALUE_ID_SEPARATOR)[-1])
+        if order_by:
+            pairs = []
+            for x in result_set:
+                pairs.extend([0, x])
+            if pairs:
+                filtered_res_key = 'filtered_result-{}'.format(uuid.uuid1())
+                await db.zadd(filtered_res_key, pairs[0], pairs[1], *pairs[2:])
+                sorted_result = await db.sort(
+                    filtered_res_key,
+                    by='{}:*->{}'.format(cls.__name__, order_by),
+                    alpha=True,
+                    asc=direction
+                )
+                # Delete the temp store
+                await db.delete(filtered_res_key)
+                return sorted_result
```

Comment:
This comment was made on a previous version but is still relevant:
https://github.com/paxos-bankchain/subconscious/pull/25#discussion_r116567598